src/main/java/baseball/service/BaseBallGameService.java

```diff
@@ -0,0 +1,105 @@
+package baseball.service;
+
+import baseball.model.Computer;
+import baseball.model.GameResult;
+import baseball.model.Hint;
+import baseball.model.Player;
+import baseball.utils.Validator;
+import baseball.utils.constants.GameCommand;
+import baseball.utils.constants.GameNumber;
+import baseball.view.InputView;
+import baseball.view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+public class BaseBallGameService {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final Validator validator;
+
+    public BaseBallGameService(InputView inputView, OutputView outputView, Validator validator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.validator = validator;
+    }
+
+    public Player savePlayerNumber() {
+        return new Player(convertStringToNumber(requestPlayerNumber()));
+    }
+
+    private List<Integer> convertStringToNumber(String input) {
+        List<Integer> result = new ArrayList<>();
+        for (char c : input.toCharArray()) {
+            result.add(c - '0');
+        }
+        return result;
+    }
+
+    private String requestPlayerNumber() {
+        String input = inputView.printRequestInput();
+        validator.validate(input);
+        return input;
+    }
+
+    public void playGame() {
+        Computer computer = new Computer();
+        while(true) {
+            Player player = savePlayerNumber();
+            GameResult gameResult = numberComparator(computer, player);
+            Hint hint = generateHint(gameResult);
+            outputView.printHintMsg(hint);
+            if(gameResult.getStrike() == GameNumber.NUMBER_MAX_SIZE.getNumber()) {
+                outputView.printEndMsg();
+                break;
+            }
+        }
+    }
+    public boolean validRestartCommand(String restartCommand) {
+        if(Objects.equals(restartCommand, GameCommand.RESTART.getCommand())) {
+            return true;
+        } else if(Objects.equals(restartCommand, GameCommand.END.getCommand())) {
+            return false;
+        } else {
+            throw new IllegalArgumentException("1 또는 2를 입력하여 다시 시작하거나 종료해주세요 !");
+        }
+    }
```

Comment:
얼리 리턴 패턴 이라는 것이 있는데 한번 참고해보시면 좋을 것 같습니다! 
https://velog.io/@orodae/얼리-리얼-패턴Early-Return-Pattern

추가로 getCommand를 하는 것이 아니라 GameCommand 객체에 isRestart인지 isEnd인지 물어보는 방식도 좋을 것 같습니다.
https://skyblue300a.tistory.com/18 
해당 글을 읽어보시면 좋을 듯 합니다!

Reply:
얼리리턴 패턴에 대해서 주신 링크에 대해서 확인 해 보았는데요 얼리리턴이라는게 넓은 범위에서부터 막아가면서 더 많은 로직을 지나지 않아도 되게 만든다는 뜻인가요 ?

getter 사용을 지양해야한다는 글을 본적이 있긴했는데 지키지 못했네요..
Enum에 true와 false도 넣어봤는데 전 이런 방식이 더 괜찮을거라고 생각했는데 아니였군요 글 참고해서 수정해보도록 하겠습니다 ! 감사합니다

Reply:
제가 이해한 얼리리턴은 else문을 지양하는 데에 초점이 있습니다! 
else 문이 많이지면 점점 어떤 조건에 의해 제어되는지 파악하기 힘들 수도 있다는 점이 있다고 생각합니다! 

해서 만약 제가 코드를 작성한다면 아래처럼 작성해볼 듯 합니다! 
```suggestion
    public boolean validRestartCommand(GameCommand restartCommand) {
        if(restartCommand.isRestart()) {
            return true;
        }
       if(restartCommand.isEnd()) {
            return false;
        }
       throw new IllegalArgumentException("1 또는 2를 입력하여 다시 시작하거나 종료해주세요 !");
    }
```

이러면 어떤 조건어 어떤 값이 리턴되는지 좀더 짧고 명확하게 표현할 수 있지 않을까 생각합니다!

Reply:
감사합니다 else 문을 최대한 줄이는게 가독성이 더 뛰어나고 파악하기 쉽겠네요 참고해서 해보도록 하겠습니다 감사합니다 !